src/js/index.js

```diff
@@ -1,22 +1,54 @@
 import React from 'react';
-import { render } from 'react-dom';
-import { App } from './App.jsx';
 import getMuiTheme from 'material-ui/styles/getMuiTheme';
 import injectTapEventPlugin from 'react-tap-event-plugin';
 import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import { render } from 'react-dom';
+import { Router, browserHistory } from 'react-router';
+
+import AboutRoute from './routes/about';
+import All from './routes/all';
+import App from './components/App.jsx';
+import ApiRoute from './routes/api';
 import { globalStyles } from './globalStyles.js';
+import LegendRoute from './routes/legend';
+import ResultRoute from './routes/result';
+
+// Apply app wide styles here (http://www.material-ui.com/#/customization/themes)
+const muiTheme = getMuiTheme(globalStyles);
+
+// react-router routes main config
+const routesConfig = {
+  path: '/',
+  // required - always load
+  component: App,
+  // load when on index only, on-demand
+  getIndexRoute(partialNextState, callback) {
+    require.ensure([], (require) => {
+      callback(null, {
+        component: require('./components/Search.jsx').default,
+      });
+    });
+  },
+  // loaded later on-demand
+  // https://github.com/ReactTraining/react-router/blob/master/docs/guides/DynamicRouting.md#dynamic-routing
```

Comment:
Yep, cool. It probably is. After reviewing it more, I can't see this particular piece of the larger picture growing excessively.